content/blog/eng/careful-with-that-constant-developer.md

```diff
@@ -0,0 +1,458 @@
+
+```
+MAXIMUM_MIDICHLORIAN_COUNT = 20000 
+```
+
+It is different isn't it? By reading the name, you are immediately able to understand what its value is about. MAXIMUM_MIDICHLORIAN_COUNT is what we call a *named constant* or simply a constant.  
+So, why do we use constants? As you have just seen, it's a matter of readability. But there's more.  
+Let's say you have a function or a method that takes an argument and does something with it. For example, the argument could be a number and, based on its value, the function could behave differently.  
+Take a look at the following code:
+
+```php
+<?php
+declare(strict_types=1);
+
+class JediTest
+{
+    public static function checkMidichlorianCount(int $midichlorianCount): string
+    {
+        if($midichlorianCount > 20000) {
+            return "The chosen one, the boy may be!"
+        }
+        return "Just a regular Jedi...";
+    }
+}
+```
+
+Do you remember? 20000 was the value of MAXIMUM_MIDICHLORIAN_COUNT. Therefore we should have used that named constant insted of the literal one.  
+Again, why do we use constants? As you have just seen, constants can be very useful to represent boundaries or edge cases.  
+And there would be more, but let's just focus on the fact that we assign certain values to immutable variables instead of simple variables. Why is that? Because we need the programming language to protects us from the possibility that our value is changed.  
+Therefore, in the end, why do we use constants? For sure it's a matter of safety.  
+Programming languages have their own way to identify something that can't be changed, for example C++ uses `const` and Java uses `final`.  
+It may not seem like it, but there are a few interesting things to say about constants in PHP.  
+
+# Global scope constants
+
+The easiest way to set a constant is using [global scope constants](https://www.php.net/manual/en/language.constants.php).  
+With the keyword `define`, you can do something like that
+
+```php
+<?php
+declare(strict_types=1);
+
+define('LIGHT_SIDE_OF_THE_FORCE', 'light');
+define('DARK_SIDE_OF_THE_FORCE', 'dark');
+
+echo LIGHT_SIDE_OF_THE_FORCE;
+// light
+echo DARK_SIDE_OF_THE_FORCE;
+// dark
+```
+ 
+However, let's say you don't need a global scope constant since you have a class hierarchy. Therefore, you need a constant that doesn't belong to a particular instance so that it can be shared between all the instances of its object.  
+Something you would write, for instance in Java, using a `static final` field.  
+PHP, at present (version 8.0.2) allows using [`final`](https://engineering.facile.it/blog/eng/from-zero-to-infinite-the-final-keyword/) only for classes and methods and that's why class constants exist.  
+
+# Class constants
+
+According to [PHP documentation](https://www.php.net/manual/en/language.oop5.constants.php), you can define constants on a per-class basis by using the keyword `const`.  
```

Comment:
Wouldn't an article here be more correct? @mar-gil? 
```suggestion
According to the [PHP documentation](https://www.php.net/manual/en/language.oop5.constants.php), you can define constants on a per-class basis by using the keyword `const`.  
```

Reply:
Well, both versions are correct.
Let's say that without the article, it has a more generale meaning, i.e. "secondo la documentazione di PHP"
Otherwise, with the article, it has a more specific meaning, the stress is on **this** documentation (the linked one).
Both versions are fine for me, what do you think @Jean85 ?

Reply:
Oh I didn't know about the difference. I would say that both are fine then.